nova/cmd/manage.py

```diff
@@ -2250,11 +2253,14 @@ def _get_rp_uuid_for_host(ctxt, host):
                 cctxt, host)
 
             if len(nodes) > 1:
-                # This shouldn't happen, so we need to bail since we
-                # won't know which node to use.
-                raise exception.TooManyComputesForHost(
-                    num_computes=len(nodes), host=host)
-            return nodes[0].uuid
+                if 'ironic' not in host:
+                    # This shouldn't happen, so we need to bail since we
+                    # won't know which node to use.
```

Comment:
This comment has gotten a bit out of touch now, and "This" is not a clear reference to me. How about:
```suggestion
                    # Multiple nodes for non-ironic hosts shouldn't happen, so
                    # we need to bail since we won't know which node to use.
```